Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -143,14 +143,15 @@ private extension ProductStore {
                                excludedProductIDs: excludedProductIDs) { [weak self] result in
                                 switch result {
                                 case .failure(let error):
-                                    onCompletion(error)
+                                    onCompletion(.failure(error))
                                 case .success(let products):
                                     if pageNumber == Default.firstPageNumber && shouldDeleteStoredProductsOnFirstPage {
                                         self?.deleteStoredProducts(siteID: siteID)
                                     }
 
                                     self?.upsertStoredProductsInBackground(readOnlyProducts: products) {
-                                        onCompletion(nil)
+                                        let hasNextPage = products.count == pageSize
+                                        onCompletion(.success(hasNextPage))
```

Comment:
Yep. All these `self?` usage scares me a bit. 😄 It's probably unlikely since this is running on the same thread. I think. 🤔

Reply:
yup thanks for bringing this up 😄 updated in https://github.com/woocommerce/woocommerce-ios/pull/2748/commits/37fbfe7bce3667ee25ffda170a17351f38eb2718